.github/workflows/checkstyle.yml

```diff
@@ -0,0 +1,19 @@
+name: Checkstyle
+
+on:
+  push:
+    branches: main
+  pull_request:
+
+jobs:
+  checkstyle:
+    name: Checkstyle
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: dbelyaev/action-checkstyle@v0.9.4
+        with:
+          github_token: ${{ secrets.GITHUB_TOKEN }}
+          reporter: github-pr-review
+          level: warning
```

Comment:
Shouldn't this be an error?

Reply:
Okay, I've done some limited testing here and I'm not sure it's actually affecting the pass/fail result here. Seemed to pass regardless of whether it was set to warning or error... maybe something to do with how checkstyle interacts with reviewdog?
`fail_on_error` seems to be the property of importance here...